app/models/folder_searcher.rb

```diff
@@ -0,0 +1,31 @@
+class FolderSearcher
+
+  def initialize(user, params: {})
+    @user = user
+    @params = params
+  end
+
+  def search
+    query = 'attachment_file_name ILIKE :search OR description ILIKE :search'
+    @folder = Attachment.where(query, search: "%#{search_term.strip}%").map do |a|
```

Comment:
you're assigning attachments to a var called folder, is this doing what you want?

Why not query unlocked resources only if a user can't download them, otherwise query all the things.

Reply:
@TheDudeWithTheThing I also felt odd about using *folder* for that assignment, however assigning it that way does help keep consistent when I want the view to show actual folders in the table versus when a search is made and the same table returns files (attachments). Do you foresee a problem with this approach going forward?

I may not be clear on your second point. From what I understand, I am returning only files that are appropriate based on the lock status of the file's parent folder and the current_user's permissions. Are you saying that this logic should not be inside the map loop?

Reply:
how is a folder an attachment though?  I feel like I'm missing something.

if you were doing `a.folder` on line 12 then it would make sense

2nd point

You should be able to do this in the query

my quick version without validating anything works
```
@folder = Folder.joins(:attachments).where(query)
return @folder if @user.dl_locked_resources?

@folder = @folder.where(locked: false)
```

throw scopes on Folder for locked / unlocked and this is even simpler. Let me know if you have questions!

Reply:
@TheDudeWithTheThing and @markottaviani 
I believe I understand the notes above, however let me outline how I interpreted this ticket as I may be at fault:
- Upon visiting `folder#index` a user will be presented with a table of all `folders`
- A user has the option to search for `attachments` within the folders (search is done by file name or description)
- Once a search is performed the results are returned on the same page (`folder#index`)
- The results are not `folders` but `attachments` and the same table that displayed the `folders`  before the search will now display the appropriate `attachments`

I agree that `@folders` within the `FolderSearcher` class is not appropriate and should be changed to `attachments`. That being said I wonder if it is best to change `FolderSearcher` to `AttachmentSearcher`.

 I have been attempting to use `joins` however I am running into an error that states:
`Can not eagerly load the polymorphic association`. However, I may not be building the query correctly.

This may be best discussed during our meeting, but wanted to add my thoughts here as well. Thanks for helping me out with this! :)

Reply:
Oh no we forgot to discuss! I can look at this today.